### PR TITLE
Unit tests 2

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
More fixed tests, pending Hugo's fixes

### CODE REVIEW

File: src/controller/tests/TestWriteChunking.cpp [hunk 1]
```diff
@@ -185,7 +197,7 @@ CHIP_ERROR TestAttrAccess::Write(const app::ConcreteDataAttributePath & aPath, a
  * resulting in N+1 attributes fitting in a write request chunk.
  *
  * This will cause all the various corner cases encountered of closing out the various containers within the write request and
- * thoroughly and definitely validate those edge cases.
+ * thoroughly and definitely validate thoses edge cases.
```
Comment (paul-szczepanek-arm) [inline src/controller/tests/TestWriteChunking.cpp:206]: rebase artefact?

File: src/controller/tests/TestEventChunking.cpp [hunk 2]
```diff
@@ -286,14 +286,20 @@ void GenerateEvents(nlTestSuite * apSuite, chip::EventNumber & firstEventNumber,
 }
 
 } // namespace
-/*
```
Comment (paul-szczepanek-arm) [inline src/controller/tests/TestEventChunking.cpp:290]: why did this get lost?

File: src/app/util/mock/attribute-storage.cpp [hunk 3]
```diff
@@ -279,7 +279,8 @@ CHIP_ERROR ReadSingleMockClusterData(FabricIndex aAccessingFabricIndex, const Co
         (emberAfGetServerAttributeIndexByAttributeId(aPath.mEndpointId, aPath.mClusterId, aPath.mAttributeId) != UINT16_MAX);
 
     ChipLogDetail(DataManagement, "Reading Mock Endpoint %" PRIx32 "Mock Cluster %" PRIx32 ", Field %" PRIx32 " is dirty",
-                  aPath.mEndpointId, aPath.mClusterId, aPath.mAttributeId);
+                  (long unsigned int) aPath.mEndpointId, (long unsigned int) aPath.mClusterId,
```
Comment (paul-szczepanek-arm) [inline src/app/util/mock/attribute-storage.cpp:282]: please use fixed formats

File: src/platform/openiotsdk/SystemPlatformConfig.h [hunk 4]
```diff
@@ -33,6 +33,7 @@
 #define CHIP_SYSTEM_CONFIG_MBED_LOCKING 0
 #define CHIP_SYSTEM_CONFIG_NO_LOCKING 0
 #define CHIP_SYSTEM_CONFIG_CMSIS_RTOS_LOCKING 1
+#define CHIP_SYSTEM_CONFIG_POOL_USE_HEAP 1
```
Comment (paul-szczepanek-arm) [inline src/platform/openiotsdk/SystemPlatformConfig.h:36]: if it's needed for tests, please put it in a config only used by the test build

File: src/platform/openiotsdk/PlatformManagerImpl.cpp [hunk 5]
```diff
@@ -109,8 +109,9 @@ CHIP_ERROR PlatformManagerImpl::_PostEvent(const ChipDeviceEvent * eventPtr)
 
 void PlatformManagerImpl::HandlePostEvent()
 {
+    uint32_t flags = 0;
     /* handle all events */
-    while (osMessageQueueGetCount(mQueue))
+    while (osMessageQueueGetCount(mQueue) && !(flags & kTaskStopEventFlag))
```
Comment (paul-szczepanek-arm) [inline src/platform/openiotsdk/PlatformManagerImpl.cpp:114]: why can you not post to a queue when it's stopping?
